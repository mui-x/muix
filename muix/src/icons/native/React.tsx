import createSvgIcon from './lib/create-svg-icon'

export default createSvgIcon(
  'react',
  'React',
  true
)