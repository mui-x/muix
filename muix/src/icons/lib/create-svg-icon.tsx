import React from 'react';
import SvgIcon from 'reactxx-mui-web/SvgIcon/SvgIcon';

const createSvgIcon = (data: string, displayName: string, isMdi:boolean) => {
    const Icon: React.SFC = props => <SvgIcon {...props} >
        <path d={data} />
    </SvgIcon >

    const res: any = Icon
    res.displayName = displayName;
    res.muiName = 'SvgIcon';

    return res;
}

export default createSvgIcon;