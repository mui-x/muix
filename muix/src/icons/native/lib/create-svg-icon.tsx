import React from 'react';

const createSvgIcon = (data: string, displayName: string, isMdi:boolean) => {
    return null
}

export default createSvgIcon;