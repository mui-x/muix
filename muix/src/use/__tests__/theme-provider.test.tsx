/** @jsx platform.createElement */
import React from 'react'

import { platform } from "reactxx-sheeter"
import { fireEvent } from 'react-testing-library'

import { ThemeProvider, useTheme } from "reactxx-use"

import { initPlatform, mount } from "./init-platform"
import { compCreator, Theme, Shape } from "./lib"

const secondThemeCreator = () => ({
  primaryColor: 'red',
  p1Prop: 'secondThemeProp'
})

describe("THEME PROVIDER", () => {

  const ToggleTheme: React.SFC = () => {
    const [theme, setTheme] = useTheme()
    const secondTheme = React.useRef(secondThemeCreator())
    return <span data-testid='toggle' onClick={() => {
      setTheme(theme === secondTheme.current ? null : secondTheme.current)
    }} />
  }

  const App: React.SFC = ({ children }) => {
    const [theme, setTheme] = React.useState(null)
    return <ThemeProvider<Theme> theme={theme}>
      <ToggleTheme />
      {children}
    </ThemeProvider>
  }

  const doTest = (isWeb: boolean) => {
    beforeEach(() => {
      initPlatform(isWeb, { dataTraceFlag: 'short' })
      platform.getDefaultTheme = () => ({
        primaryColor: 'gray',
        p1Prop: 'themeProp'
      })
    })

    it('01: ', () => {
      const Comp = compCreator({ defaultSheet: { root: {} } }, null)
      const {getByTestId, container, unmount} = mount(<App>
        <Comp themedProps={theme => ({ p1: theme.p1Prop })} classNameX={theme => ({ color: theme.primaryColor }) as any} />
      </App>)
      const toggle = getByTestId('toggle')
      
      expect(container).toMatchSnapshot()
      fireEvent.click(toggle)
      //toggle.click()
      expect(container).toMatchSnapshot()
      fireEvent.click(toggle)
      expect(container).toMatchSnapshot()
      unmount()
    })

  }

  describe("## NATIVE ##", () => doTest(false))
  describe("## WEB ##", () => doTest(true))

})

