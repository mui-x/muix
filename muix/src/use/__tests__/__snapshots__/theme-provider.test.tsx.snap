// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`THEME PROVIDER ## NATIVE ## 01:  1`] = `
<div>
  <span
    data-testid="toggle"
  />
  <div
    data-trace=""
    p1="themeProp"
    style="color: gray;"
  >
    1
  </div>
</div>
`;

exports[`THEME PROVIDER ## NATIVE ## 01:  2`] = `
<div>
  <span
    data-testid="toggle"
  />
  <div
    data-trace=""
    p1="secondThemeProp"
    style="color: red;"
  >
    2
  </div>
</div>
`;

exports[`THEME PROVIDER ## NATIVE ## 01:  3`] = `
<div>
  <span
    data-testid="toggle"
  />
  <div
    data-trace=""
    p1="themeProp"
    style="color: gray;"
  >
    3
  </div>
</div>
`;

exports[`THEME PROVIDER ## WEB ## 01:  1`] = `
<div>
  <span
    data-testid="toggle"
  />
  <div
    class="a"
    data-trace="
.a { color:gray }"
    p1="themeProp"
  >
    1
  </div>
</div>
`;

exports[`THEME PROVIDER ## WEB ## 01:  2`] = `
<div>
  <span
    data-testid="toggle"
  />
  <div
    class="b"
    data-trace="
.b { color:red }"
    p1="secondThemeProp"
  >
    2
  </div>
</div>
`;

exports[`THEME PROVIDER ## WEB ## 01:  3`] = `
<div>
  <span
    data-testid="toggle"
  />
  <div
    class="a"
    data-trace="
.a { color:gray }"
    p1="themeProp"
  >
    3
  </div>
</div>
`;
